### DISCUSSION TITLE
What about _Component.svelte (shadow) endpoints?

### DISCUSSION THREAD

**Discussion by lkj4** [sveltejs/kit#4896] [Ideas]
Pretty much the title. So I'd like to create shadow endpoints for `_Component.svelte` which are not page endpoints. 

Example: A `_ShowDetailModal.svelte` is a component which is shown on a page but is not a dedicated endpoint. Can this component have also a dedicated shadow endpoint which is fetched automagically when the component is mounted?

**Comment by mrkishi** [comment]
The thing with per-component data-fetching is that they can't be done efficiently or magically without changing Svelte's internals.

Given this snippet:

```svelte
{#if show}
    This is my component: <MyComponent {someProp} />
{/if}
```

What would happen when `show` turns `true`? Svelte wouldn't even know `MyComponent` has some `async` work to do.

Likewise, how would SSR work? Svelte SSR is currently synchronous, so we would need the data prior to rendering that component. But we'd also need to render it first to know what `someProp` is and if `MyComponent` is even used!

**Reply by lkj4** [reply]
Makes sense, thanks!
